tests/Model/SearchIndexInputTest.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace MongoDB\Tests\Model;
+
+use MongoDB\BSON\Serializable;
+use MongoDB\Exception\InvalidArgumentException;
+use MongoDB\Model\SearchIndexInput;
+use MongoDB\Tests\TestCase;
+
+class SearchIndexInputTest extends TestCase
+{
+    public function testConstructorIndexDefinitionMustBeDefined(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->expectExceptionMessage('Required "definition" document is missing from search index specification');
+        new SearchIndexInput([]);
+    }
+
+    public function testConstructorIndexDefinitionMustBeADocument(): void
```

Comment:
I assume this was based on `testConstructorShouldRequireKeyToBeArrayOrObject` from `IndexInputTest`. Would it make more sense to rewrite both to use the base TestCase's `provideInvalidDocumentValues()` data provider for completeness?